src/App.tsx

```diff
@@ -14,11 +15,126 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  sortBy: string,
+  isRendered: boolean,
+  isReversed: boolean,
+  selectedLength: number,
+};
+
+class App extends React.PureComponent<{}, State> {
+  state = {
+    goods: [...goodsFromServer],
+    isRendered: false,
+    isReversed: false,
+    sortBy: '',
+    selectedLength: 1,
+  };
+
+  showList = () => (
+    this.setState({
+      isRendered: true,
+    }));
+
+  reverseGoods = () => (
+    this.setState((state) => ({ isReversed: !state.isReversed })));
+
+  sortByName = () => (
+    this.setState({
+      sortBy: 'name',
+    }));
+
+  sortByLength = () => (
+    this.setState({
+      sortBy: 'length',
+    }));
+
+  resetGoods = () => (
+    this.setState({
+      isReversed: false,
+      selectedLength: 1,
+      sortBy: '',
+    })
+  );
+
+  render() {
+    const {
+      goods, isRendered, isReversed, sortBy, selectedLength: itemLength,
```

Comment:
are you sure you need selectedLength: itemLength ?

Reply:
Wanted to do advanced tasks with Select Button ( All the previous buttons
  should now show only goods having length >= than the selected value).  But it did not work. And I forgot to delete what I don't use